### DISCUSSION TITLE
Does the dynamic configuration reload functionality work with SSH repositories?

### DISCUSSION THREAD

**Discussion by brandonp42** [containers/fetchit#314] [Q&A] [answered]
All of the examples using dynamic configuration reload (https://fetchit.readthedocs.io/en/latest/methods.html) seem to use HTTP. Does it support SSH?

Thank you!

**Comment by cooktheryan** [comment, marked as answer]
Currently the engine only support HTTP based on how it is written a change
could be introduced that would allow for the user to specify SSH or HTTP.

**Reply by brandonp42** [reply]
Thank you, I'm not sure if I will have time to attempt a PR for this but might depending on how successful I am at getting my team to adopt FetchIt.

**Comment by cooktheryan** [comment]
Ill put a slot on my calendar to look into it a bit next week.

**Reply by brandonp42** [reply]
That would be awesome. So far in my testing SSH seems to be best from a security perspective since its easy to keep the ssh key out of the configuration repository. If there was a way to put a Bitbucket/GitHub token in a file outside the configuration file that would help but my security team doesn't want us using HTTP tokens so I don't think I would be able to use that.

**Reply by brandonp42** [reply]
You know, just trying to share where I'm coming from with this.

**Reply by cooktheryan** [reply]
@brandonp42 one solution we looked at was using podman secrets

**Reply by cooktheryan** [reply]
so looking at this I think the solution would be to cut a release at the current time then go in and make the following modification to the config reload engine. We will need to allow for it to do a shallow clone of a git repo and accept a path + filename. Once it has those assets do a move of the file to the expected location.
